lms/djangoapps/bookmarks/models.py

```diff
@@ -36,23 +49,73 @@ def create(cls, bookmark_data):
         Raises:
             ItemNotFoundError: If no block exists for the usage_key.
         """
-        usage_key = bookmark_data.pop('usage_key')
+        data = copy.deepcopy(data)
```

Comment:
I would advise against `deepcopy` in general. It can be very performance intensive. Looking at the code below, I think you only need a shallow copy to protect the caller from the mutation that you're doing (`dict(data)`), or you could just write in the API that this method may mutate the argument.